test/databases/block_database.cpp

```diff
@@ -236,8 +235,10 @@ BOOST_AUTO_TEST_CASE(block_database__test)
 
     // no metadata for missing blocks
     instance.get_header_metadata(block4a.header());
-    BOOST_REQUIRE_EQUAL(block4b.header().metadata.candidate, block_state::missing);
-    BOOST_REQUIRE_EQUAL(block4b.header().metadata.confirmed, block_state::missing);
+
+    // BUGBUG: bool vs. state comparisons.
+    ////BOOST_REQUIRE_EQUAL(block4b.header().metadata.candidate, block_state::missing);
```

Comment:
@evoskuil ouch! :)

We can replace these with

`BOOST_REQUIRE(!block4b.header().metadata.candidate)`
`BOOST_REQUIRE(!block4b.header().metadata.confirmed)`

I verified the change locally, just to be sure.